core-web/libs/sdk/client/src/lib/client/content/methods/get-collection/get-collection.ts

```diff
@@ -0,0 +1,285 @@
+import { sanitizeQueryForContentType } from '../../shared/utils';
+
+/**
+ * 'GetCollection' Is a Typescript class that provides the ability build a query to get a collection of content.
+ *
+ * @export
+ * @class GetCollection
+ */
+export class GetCollection {
+    #page = 1;
+    #limit = 10;
+    #depth = 0;
+    #render = false;
+    #sortBy?: SortByArray;
+    #contentType: string;
+    #defaultQuery: Equals;
+    #query?: Equals;
+    #rawQuery?: string;
+
+    private serverUrl: string;
+    private requestOptions: Omit<RequestInit, 'body' | 'method'>;
+
+    /**
+     * This method returns the sort query in this format: field order, field order, ...
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get sort() {
+        return this.#sortBy?.map((sort) => `${sort.field} ${sort.order}`).join(',');
+    }
+
+    /**
+     * This method returns the offset of the content to fetch
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get offset() {
+        // This could end in an empty response
+        return this.#limit * (this.#page - 1);
+    }
+
+    /**
+     * This method returns the url to fetch the content
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get url() {
+        return `${this.serverUrl}${CONTENT_API_URL}`;
+    }
+
+    /**
+     * This method returns the current query built
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get currentQuery() {
+        return this.#query ?? this.#defaultQuery;
+    }
+
+    constructor(
+        requestOptions: Omit<RequestInit, 'body' | 'method'>,
+        serverUrl: string,
+        contentType: string
+    ) {
+        this.requestOptions = requestOptions;
+        this.serverUrl = serverUrl;
+
+        this.#contentType = contentType;
+
+        // We need to build the default query with the contentType field
+        this.#defaultQuery = new QueryBuilder().field('contentType').equals(this.#contentType);
+    }
+
+    /**
+     * This method allows to filter the content by language.
+     *
+     * @param {number} language
+     * @return {*}  {this}
+     * @memberof GetCollection
+     */
+    language(language: number): this {
+        this.#query = this.currentQuery.field('languageId').equals(language.toString());
+
+        return this;
+    }
+
+    /**
+     * This method allows you to fetch the render version of the content
+     *
+     * @param {boolean} render
+     * @return {*}  {this}
+     * @memberof GetCollection
+     */
+    render(render: boolean): this {
+        this.#render = render;
+
+        return this;
+    }
+
+    /**
+     * This method allows you to sort the content by field
+     *
+     * @param {SortByArray} sortBy
+     * @return {*}  {this}
+     * @memberof GetCollection
+     */
```

Comment:
We will do this later, but here is an example of the doc of `firestore` for the sort method:

```
/**
  * Creates a {@link QueryOrderByConstraint} that sorts the query result by the
  * specified field, optionally in descending order instead of ascending.
  *
  * Note: Documents that do not contain the specified field will not be present
  * in the query result.
  *
  * @param fieldPath - The field to sort by.
  * @param directionStr - Optional direction to sort by ('asc' or 'desc'). If
  * not specified, order will be ascending.
  * @returns The created {@link QueryOrderByConstraint}.
  */
```